phosphor-auditlog/auditLogParser.cpp

```diff
@@ -0,0 +1,463 @@
+                /* No more records, something is confused! */
+                lg2::error(
+                    "Record count ({NRECS}) and records out of sync ({ITER})",
+                    "NRECS", nRecords, "ITER", iter);
+                throw sdbusplus::xyz::openbmc_project::Common::Error::
+                    InternalFailure();
+
+                break;
+            case -1:
+            default:
+                /* Error */
+                lg2::error("Failed on record number={ITER}", "ITER", iter);
+                throw sdbusplus::xyz::openbmc_project::Common::Error::
+                    InternalFailure();
+                break;
+        }
+    }
+}
+
+/**
+ * @brief Parses general audit entry into JSON format
+ * @details Used with audit entries without specific handling. Text of audit
+ * log message is written as-is.
+ */
+void ALParser::fillAuditEntry(nlohmann::json& parsedEntry)
+{
+    parsedEntry["MessageId"] = "OpenBMC.0.5.AuditLogEntry";
+
+    /* MessageArgs: msg */
+    auto recMsg = auparse_get_record_text(au);
+    nlohmann::json messageArgs = nlohmann::json::array({recMsg});
+
+    parsedEntry["MessageArgs"] = std::move(messageArgs);
+}
+
+/**
+ * @brief Strips '"' from beginning and end of value field
+ */
+inline std::string getValue(std::string fieldText)
+{
+    if (fieldText.starts_with('\"'))
+    {
+        auto endQuote = fieldText.find('\"', 1);
+
+        if (endQuote != std::string::npos)
+        {
+            return fieldText.substr(1, endQuote - 1);
+        }
+    }
+
+    return fieldText;
+}
+
+/**
+ * @brief Parses AUDIT_USYS_CONFIG audit entry into JSON format
+ * @details Expected fields from audit log entry are split into MessageArgs
+ */
+void ALParser::fillUsysEntry(nlohmann::json& parsedEntry)
+{
+    parsedEntry["MessageId"] = "OpenBMC.0.5.AuditLogUsysConfig";
+
+    nlohmann::json messageArgs = nlohmann::json::array();
+
+    /* Map expected fields to MessageArgs index.
+     * Audit records contain fields not returned for admin use. E.g. the pid of
+     * the auditd daemon that recorded the entry is part of the record.
+     */
+    std::map<std::string, int>::const_iterator mapEntry;
+    const std::map<std::string, int> msgArgMap({{"type", 0},
+                                                {"op", 1},
+                                                {"acct", 2},
+                                                {"exe", 3},
+                                                {"hostname", 4},
+                                                {"addr", 5},
+                                                {"terminal", 6},
+                                                {"res", 7}});
+
+    /* Walk the fields and insert mapped fields into messageArgs */
+    int fieldIdx = 0;
+    int frc;
+    size_t nFields = 0; // Used to confirm all expected fields found
+    do
+    {
+        fieldIdx++;
+
+        // Can return nullptr
+        const char* fieldName = auparse_get_field_name(au);
```

Comment:
If you wanted to write unit tests against this stuff, you could wrap all the auparse calls in a wrapper class and then use a mock version of it in the test to return what you wanted.  Just an idea for the future if you get time to add tests.

Reply:
Great idea! I took a look to see if it would be quick to make this change now. I think I would have to do some restructuring to do this. I'm going to defer this to the next sprint after I get everything merged, but I want to make this change in the near-term. I would like to work on adding unit tests for 1060.